### PR TITLE
Invalidate type_options having precision without a scale

### LINKED ISSUE #1367: Partial data loss when attempting to unset scale type_options independently of precision
## Reproduce

1. Set up a Number column with "Decimal Places" = `2` and "Max Digits" = `100`.

1. Add row with a value of `1.23`.

1. Now imagine we'd like to also store the value `6.789` without loss of precision. We'll need to modify or remove the "Decimal Places" setting.

1. Edit the DB type options. Remove the `2` in the "Decimal Places" field (leaving the field blank) and save.

1. Observe a PATCH request to the columns API.

1. Expect the response to have `type_options` of `{ precision: 100, scale: null }` or `{ precision: 100 }`.

1. Observe the response to have `type_options` of `{ precision: 100, scale: 0 }`.

1. Expect the cell value within the newly-added row to still be `1.23`.

1. Observe that the value of the cell has changed to `1` because `scale` has been changed to `0`. This is especially bad because it's **data loss**.

## Notes

- The [Postgres docs on numeric types ](https://www.postgresql.org/docs/current/datatype-numeric.html) appear to indicate that Postgres supports NUMERIC with a precision value and without a scale value.

### PR DESCRIPTION
**Pull request by Anish9901**
Fixes #1367 

Setting a precision parameter of the type_options without a valid scale could result in data loss so we disallow having precision without a scale. 

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by mathemancer — CHANGES_REQUESTED** [review]
Looks good overall. I did make a suggestion in a specific comment. Please consider it.

File: mathesar/api/serializers/columns.py [hunk 1]
```diff
@@ -33,8 +38,19 @@ class TypeOptionSerializer(MathesarErrorMessageMixin, serializers.Serializer):
...
-        if attrs.get('scale', None) is not None and attrs.get('precision', None) is None:
-            attrs['precision'] = 1000
+        db_type = self.context.get('db_type', None)
+        scale = attrs.get('scale', None)
+        precision = attrs.get('precision', None)
+        if (
+            db_type == PostgresType.NUMERIC
+            and (scale is None and precision is not None)
+            or (scale is not None and precision is None)
+        ):
```
[the remark this answers is not in the document]
Comment (Anish9901) [reply mathesar/api/serializers/columns.py:48]: You are right it's an xor, I should have noticed that.

**Review by mathemancer — APPROVED** [review]
Okay, this looks good to me. Thanks!

**Comment by kgodey** [comment]
This won't auto-merge due to merge conflicts @Anish9901 @mathemancer

**Comment by mathemancer** [comment]
Thanks for the heads-up @kgodey

**Pull request merged**